packets.go

```diff
@@ -495,7 +499,7 @@ func (mc *mysqlConn) readResultOK() ([]byte, error) {
 			if len(data) > 1 {
 				pluginEndIndex := bytes.IndexByte(data, 0x00)
 				plugin := string(data[1:pluginEndIndex])
-				cipher := data[pluginEndIndex+1 : len(data)-1]
+				cipher := data[pluginEndIndex+1:]
```

Comment:
My concern was if it breaks native password auth, since it changes the behavior introduced in #494. 
But this is actually addressed in this PR by slicing the data to the specified size, so everything should be fine.